stytch/core/http/client.py

```diff
@@ -86,9 +87,28 @@ def delete(
 
 
 class AsyncClient(ClientBase):
-    def __init__(self, project_id: str, secret: str) -> None:
+    def __init__(
+        self,
+        project_id: str,
+        secret: str,
+        session: Optional[aiohttp.ClientSession] = None,
+    ) -> None:
         super().__init__(project_id, secret)
         self.auth = aiohttp.BasicAuth(project_id, secret)
+        self._external_session = session is not None
+        self._session = session or aiohttp.ClientSession()
```

Comment:
This changes the semantics to create a ClientSession only once on startup instead of per-request - while the external API is the same the behavior is not backwards compatible. Are there any cases where callers would want to use (or implicitly depend on) the previous behavior? For example, if they're creating a Stytch client per-request and have a memory leak that prevents the client from getting GC'd, then we'd turn a small memory leak into a much larger one.

Does it make sense to make this a major version bump instead?

Reply:
> Does it make sense to make this a major version bump instead?

I'm supportive of that if we want to be extra cautious.